HACKING.md

```diff
@@ -27,31 +27,35 @@ in order to make it to the first minor release.
 Pull Requests
 -------------------------------------------------
 
-### Top reasons pull-requests are rejected or delayed
+### Checklist
 
-* Code does not follow style guidlines. See [Coding Style](#coding-style).
+When submitting your pull request, please make sure you:
 
-* Pull request addresses several disparate issues. In general, smaller
-pull-requests are better because they are easier to review and stay mergeable
-longer.
+* Follow our [Coding Style](#coding-style).
 
-* Big feature is added, but it is not configurable in compile-time.
-We are striving to keep CFEngine lightweight and fast, so big new
-features should be possible to disable with
-`./configure --disable-feature` and linking to new libraries
-should be optional with `./configure --without-libfoo`.
+* Address only one issue/feature.
+  Smaller pull requests are generally better.
 
-* Messy commit log. Tidy up the commit log by squashing commits.
+* Add tests.
+  C functions should have unit tests.
+  Promise types, attributes, and functions should have acceptance tests.
 
-* Missing ChangeLog description in commit message, which is mandatory
-for new features or bugfixes to be accepted.
-See [ChangeLog Entries](#changelog-entries) for details.
+* Pay attention to review comments and CI results.
+  We have code analysis and tests which run when you submit your PR.
+  You may have to make some changes.
 
-* Code is out-of-date, does not compile, or does not pass all tests. Again,
-focused and small pull-requests are better.
+* Check that you don't have any merge conflicts.
+  (Rebase on master or target branch).
+
+* Tidy up the commit log by squashing commits.
+  Each commit should be a valid change and make sense on its own.
+  Usually a Pull Request will only have one commit.
+
```

Comment:
separate whitespace commits?

Reply:
There is a separate whitespace commit, but I squashed the 2 other commits so this makes more sense.

Reply:
No I mean in the text specify that PRs should include a separate commit for whitespace changes.